packages/dialect-react-ui/components/Chat/provider/index.tsx

```diff
@@ -0,0 +1,37 @@
+import { createContext, FunctionComponent, useContext } from 'react';
+
+// We are duplicating this type in order to have separation of concern: it may be different to what we have in props
+type ChatType = 'inbox' | 'popup' | 'vertical-slider';
```

Comment:
looks very hacky, probably composition would play better here, but I do not insist, since it would take a huuge amount of time. Might revisit it during further refactoring. wdyt?

Reply:
I just thought that `type` might be a bit hacky, and too overwhelming the component itself
By composition I meant to have smth like `ChatInbox`, `ChatPopup`, `ChatVerticalSlider`, which would have composed internally with reusable components

Reply:
Great point, thanks! I suggest to discuss/plan this during next refactor of react-ui? Wdyt?

Reply:
yes def